cmd/utils/cmd.go

```diff
@@ -216,10 +228,100 @@ func ExportAppendChain(blockchain *core.BlockChain, fn string, first uint64, las
 		writer = gzip.NewWriter(writer)
 		defer writer.(*gzip.Writer).Close()
 	}
-
+	// Iterate over the blocks and export them
 	if err := blockchain.ExportN(writer, first, last); err != nil {
 		return err
 	}
 	log.Info("Exported blockchain to", "file", fn)
 	return nil
 }
+
+// ImportPreimages imports a batch of exported hash preimages into the database.
+func ImportPreimages(db *ethdb.LDBDatabase, fn string) error {
+	log.Info("Importing preimages", "file", fn)
+
+	// Open the file handle and potentially unwrap the gzip stream
+	fh, err := os.Open(fn)
+	if err != nil {
+		return err
+	}
+	defer fh.Close()
+
+	var reader io.Reader = fh
+	if strings.HasSuffix(fn, ".gz") {
+		if reader, err = gzip.NewReader(reader); err != nil {
+			return err
+		}
+	}
+	stream := rlp.NewStream(reader, 0)
+
+	// Import the preimages in batches to prevent disk trashing
+	preimages := make(map[common.Hash][]byte)
+
+	for {
+		// Read the next entry and ensure it's not junk
+		var (
+			key common.Hash
+			val []byte
+		)
+		if err := stream.Decode(&key); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		if err := stream.Decode(&val); err != nil {
+			return err
+		}
+		if hash := crypto.Keccak256Hash(val); hash != key {
+			return fmt.Errorf("hash mismatch: have %v, want %v", hash, key)
```

Comment:
Instead of erroring out here, there are two other options:

1. skip this
2. use the calculated hash. 

Option 2 would mean that we can actually skip exporting the hashes, and only export the preimates.

Reply:
That actually makes sense TBH. Let's do 2 and drop the hash altogether.